### PR TITLE
Feature/swp module

### PR DESCRIPTION
**Pull request by oisinod**
Begins work in #5204

### CODE REVIEW

File: src/analysis/retail/priest/discipline/modules/spells/PainAndSuffering.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    ) {
+      return;
+    }
+
+    const rawHeal = healEvent.amount + (healEvent.overheal || 0);
+    const nonAmpedHealing = rawHeal / this.purgeTheWicked.effectiveIncrease;
+    const increase = healEvent.amount - nonAmpedHealing;
+    if (increase > 0) {
+      this.healing += increase * this.purgeTheWicked.dotRatios.painAndSuffering;
+    }
```
Comment (kfinch) [inline src/analysis/retail/priest/discipline/modules/spells/PainAndSuffering.tsx]: I'm pretty sure this logic is the same as the existing utility `calculateEffectiveHealing` - consider using that instead. I do see there is `calculateEffectiveDamage` below - is there something mathy here that I'm not seeing?

File: src/analysis/retail/priest/discipline/modules/spells/ThroesOfPain.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    ) {
+      return;
+    }
+
+    const rawHeal = healEvent.amount + (healEvent.overheal || 0);
+    const nonAmpedHealing = rawHeal / this.purgeTheWicked.effectiveIncrease;
+    const increase = healEvent.amount - nonAmpedHealing;
+    if (increase > 0) {
+      this.healing += increase * this.purgeTheWicked.dotRatios.throesOfPain;
+    }
```
Comment (kfinch) [inline src/analysis/retail/priest/discipline/modules/spells/ThroesOfPain.tsx]: Same as above comment, can `calculateEffectiveHealing` be used?

**Comment by kfinch** [comment]
Needs a changelog entry and to fix merge conflict, then good to merge

**Comment by oisinod** [comment]
@kfinch done! :) thanks

**Pull request merged**